akka-actor-typed/src/main/scala/akka/actor/typed/Logger.scala

```diff
@@ -41,6 +42,63 @@ object LogMarker {
 
 }
 
+/**
+ * Logging options when using `Behaviors.logMessages`.
+ */
+@DoNotInherit
+abstract class LogOptions {
+  /**
+   * User control whether messages are logged or not.  This is useful when you want to have an application configuration
+   * to control when to log messages.
+   */
+  def enabled: Boolean
+
+  /**
+   * The [[akka.event.Logging.LogLevel]] to use when logging messages.
+   */
+  def level: LogLevel
+
+  /**
+   * A [[akka.actor.typed.Logger]] to use when logging messages.
+   */
+  def logger: Logger
+}
+
+/**
+ * Factories for log options
+ */
+object LogOptions {
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  final class LogOptionsImpl private[akka] (val enabled: Boolean, val level: LogLevel, val logger: Logger)
+    extends LogOptions
+
+  /**
+   * Scala API: Create a new log options.
+   */
+  def apply(enabled: Boolean, level: LogLevel, logger: Option[Logger]): LogOptions =
```

Comment:
We often use `with` methods instead of having overloaded factory methods for all variants. Also easier to evolve in binary compatible way.

```
withEnabled
withLogLevel
withLogger
```